### ISSUE TITLE
FR: Conditionally style popovers based on active view type

### ISSUE THREAD

**Issue by nothingislost** [nothingislost/obsidian-hover-editor#57]
For example, when hovering over an image embed, the popover could auto resize itself to the exact dimensions and aspect ratio of the image (and scale down to fit the window if too large).

This would likely entail taking the workspace-leaf-content `data-type` attribute and lifting it up to the .hover-popover container. It could also require conditional changes to the interact.js handlers to do things like constrain aspect ratio depending on the view type.

**Comment by pjeby** [comment]
PDFs have this issue too; they're currently one of the very few things for which the builtin popover has a better initial experience. (due to aspect ratio + less "chrome").

**Comment by nothingislost** [comment]
Yep, this is next on my todo list. I think this could be really slick. It'll be kind of like quick preview in the macos finder.

**Comment by Obadiahcruz** [comment]
Can't freaking wait! This and your other one for external links is gonna be niceeeee @nothingislost

**Comment by nothingislost** [comment]
If you update using BRAT, the 0.7.0 beta has some of these features https://github.com/nothingislost/obsidian-hover-editor/releases/tag/0.7.0

**Comment by Obadiahcruz** [comment]
just tried it and it runs really smooth in the file explorer and editor! 

what are your thoughts for hitting a key modifier to activate the popup when only using the keyboard? like how macos does it with macOS quicklook (using space).

**Comment by nothingislost** [comment]
Addressed in https://github.com/nothingislost/obsidian-hover-editor/releases/tag/0.7.0
